app/Http/Controllers/MovieController.php

```diff
@@ -4,41 +4,53 @@
 
 use App\Http\Requests\Movie\CreateRequest;
 use App\Http\Requests\Movie\EditRequest;
+use App\Models\Actor;
+use App\Models\Genre;
 use App\Models\Movie;
 use Illuminate\Http\Request;
 
 class MovieController extends Controller
 {
     public function createForm()
     {
-        return view('movies.create');
+        $genres = Genre::all();
+        $actors = Actor::all();
+        return view('movies.create', compact('genres', 'actors'));
     }
 
     public function editForm(Movie $movie)
     {
-
-        return view('movies.edit', compact('movie'));
+        $genres = Genre::all();
+        $actors = Actor::all();
+        return view('movies.edit', compact('movie', 'genres', 'actors'));
     }
 
     public function create(CreateRequest $request)
     {
         $data = $request->validated();
         $movie = new Movie($data);
+        $user = $request->user();
+        $movie->user()->associate($user);
         $movie->save();
+        $movie->genres()->attach($data['genres']);
+        $movie->actors()->attach($data['actors']);
 
         session()->flash('success', 'Success!');
+        return redirect()->route('movie.show', ['movie' => $movie->id]);
 
-        return redirect()->route('movie.show', compact('movie'));
     }
 
     public function edit(Movie $movie, EditRequest $request)
     {
         $data = $request->validated();
         $movie->fill($data);
+
+        $movie->genres()->attach($data['genres']);
+        $movie->actors()->attach($data['actors']);
```

Comment:
тут нужно sync
```suggestion
        $movie->genres()->sync($data['genres']);
        $movie->actors()->sync($data['actors']);
```

attach добавляет только, а тебе нужно перезаписать связи